.github/workflows/back-end.yml

```diff
@@ -0,0 +1,228 @@
+            name: "Check file permissions"
+            run: |
+                test "$(find ./ -type f -not -path './.git/*' -executable)" = ""
+        -
+            name: "Find non-printable ASCII characters"
+            run: |
+                ! LC_ALL=C.UTF-8 git grep --perl-regexp --line-number -e '[^ -~]' -- '*\.php'
+
+    syntax_errors:
+        name: "1️⃣ Syntax errors"
+        runs-on: "ubuntu-22.04"
+        timeout-minutes: 5
+        steps:
+        -
+            name: "Set up PHP"
+            uses: "shivammathur/setup-php@v2"
+            with:
+                php-version: "8.1"
+                coverage: "none"
+                tools: "parallel-lint"
+        -
+            name: "Checkout repository"
+            uses: "actions/checkout@v3"
+        -
+            name: "Search for conflict markers 🐌"
+            run: |
+                ! git grep --line-number -e '^\(<<<<<<<\s\|=======\s\|=======$\|>>>>>>>\s\||||||||\s\)'
+        -
+            name: "Search for invalid complex curly syntax 🐌"
+            run: |
+                ! git grep -e '\${[A-Z_a-z]' -- '*\.php'
+        -
+            name: "Check source code for syntax errors"
+            run: "composer exec --no-interaction -- parallel-lint src/ tests/"
+
+    unit_tests:
+        name: "2️⃣ Unit and functional tests"
+        needs:
+        - "byte_level"
+        - "syntax_errors"
+        strategy:
+            #fail-fast: false
+            matrix:
+                php-version:
+                - "8.1"
+                - "8.2"
+                dependencies:
+                - "lowest"
+                - "highest"
+        runs-on: "ubuntu-22.04"
+        timeout-minutes: 5
+        steps:
+        -
+            name: "Set up PHP"
+            uses: "shivammathur/setup-php@v2"
+            with:
+                php-version: "${{ matrix.php-version }}"
+        -
+            name: "Setup problem matcher for PHPUnit"
+            run: |
+                echo "::add-matcher::${{ runner.tool_cache }}/phpunit.json"
+        -
+            name: "Checkout repository"
+            uses: "actions/checkout@v3"
+        -
+            name: "Install dependencies"
+            uses: "ramsey/composer-install@v2"
+            with:
+                dependency-versions: "${{ matrix.dependencies }}"
+        -
+            name: "Execute unit tests"
+            run: "composer run unit"
+        -
+            name: "Execute feature tests"
+            run: "composer run feature"
+        -
+            name: "Send coverage to Coveralls"
+            env:
+                COVERALLS_REPO_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+            if: "${{ env.COVERALLS_REPO_TOKEN && matrix.php-version == '8.2' && matrix.dependencies == 'highest' }}"
+            run: |
+                wget --no-verbose \
+                    "https://github.com/php-coveralls/php-coveralls/releases/download/v2.5.3/php-coveralls.phar"
+                #php ./php-coveralls.phar --verbose
```

Comment:
Add coverage in `phpunit.xml` and remove comment from here.
https://coveralls.io/github/PrinsFrank/MRTD